core/src/commonMain/kotlin/com/powersync/PowerSyncDatabaseFactory.kt

```diff
@@ -0,0 +1,29 @@
+package com.powersync
+
+import com.powersync.db.PowerSyncDatabaseImpl
+import com.powersync.db.schema.Schema
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import co.touchlab.skie.configuration.annotations.DefaultArgumentInterop
+
+public const val DEFAULT_DB_FILENAME: String = "powersync.db"
+
+/**
+ * Use this to create a [PowerSyncDatabase]
+ */
+@OptIn(DelicateCoroutinesApi::class)
+@DefaultArgumentInterop.Enabled
+public fun createPowerSyncDatabase(
+    factory: DatabaseDriverFactory,
+    schema: Schema,
+    dbFilename: String = DEFAULT_DB_FILENAME,
+    scope: CoroutineScope = GlobalScope
+): PowerSyncDatabase {
+    return PowerSyncDatabaseImpl(
```

Comment:
For context, why is the `createPowerSyncDatabase` required? Could user's not just construct the client as on line 23?

Reply:
Are you asking if we want to users to use `PowerSyncDatabaseImpl` to create the DB? I guess they could but I think that could be an anti-pattern in Kotlin. I'm not sure you're meant to expose implementations to the public. Based on the feedback in the audit this appears to be the way to do it.

Reply:
Also note I've renamed `createPowerSyncDatabase` to `PowerSyncDatabase` as this seems to be the way to do it in Kotlin w.r.t a factory generating an instance of a class

Reply:
I like the latest changes